gamemodes/terrortown/gamemode/client/cl_help/populate_appearance.lua

```diff
@@ -0,0 +1,575 @@
+local materialIcon = Material("vgui/ttt/vskin/helpscreen/appearance")
+
+local function PopulateGeneralPanel(parent)
+	local form = CreateForm(parent, "header_global_color")
+
+	form:MakeHelp({
+		label = "help_color_desc"
+	})
+
+	-- store the reference to the checkbox in a variable
+	-- because the other settings are enabled based on
+	-- the state of this checkbox
+	local enbColor = form:MakeCheckBox({
+		label = "label_global_color_enable",
+		initial = appearance.ShouldUseGlobalFocusColor(),
+		OnChange = function(_, value)
+			appearance.SetUseGlobalFocusColor(value)
+		end,
+		default = false
+	})
+
+	form:MakeColorMixer({
+		label = "label_global_color",
+		initial = appearance.GetFocusColor(),
+		OnChange = function(_, color)
+			appearance.SetFocusColor(color)
+		end,
+		master = enbColor
+	})
+
+	form:MakeHelp({
+		label = "help_scale_factor"
+	})
+
+	form:MakeSlider({
+		label = "label_global_scale_factor",
+		min = 0.1,
+		max = 3,
+		decimal = 1,
+		initial = appearance.GetGlobalScale(),
+		OnChange = function(_, value)
+			appearance.SetGlobalScale(value)
+		end,
+		default = appearance.GetDefaultGlobalScale()
+	})
+end
+
+local function PopulateHUDSwitcherPanelSettings(parent, currentHUD)
+	parent:Clear()
+
+	parent:MakeHelp({
+		label = "help_hud_special_settings"
+	})
+
+	for key, data in pairs(currentHUD:GetSavingKeys() or {}) do
+		if data.typ == "color" then
+			parent:MakeColorMixer({
+				label = data.desc or key,
+				initial = currentHUD[key],
+				OnChange = function(_, color)
+					currentHUD[key] = color
+
+					if isfunction(data.OnChange) then
+						data.OnChange(currentHUD, color)
+					end
+				end,
+				showAlphaBar = true,
+				showPalette = true
+			})
+
+
+		elseif data.typ == "number" then
+			parent:MakeSlider({
+				label = data.desc or key,
+				min = data.min or 0.1,
+				max = data.max or 4,
+				decimal = data.decimal or 1,
+				initial = math.Round(currentHUD[key] or 1, 1),
+				default = data.default,
+				OnChange = function(_, value)
+					value = math.Round(value, 1)
+
+					if value ~= math.Round(currentHUD[key], 1) then
+						if isfunction(data.OnChange) then
+							data.OnChange(currentHUD, value)
+						end
+
+						currentHUD[key] = value
```

Comment:
This is inconsistent to the color picker:
There the value is already written to the HUD, before calling the OnChange function